lib/stages/timeDownsample.cpp

```diff
@@ -127,13 +136,21 @@ void timeDownsample::main_thread() {
 
             // Accumulate integration totals
             output_frame.fpga_seq_total += frame.fpga_seq_total;
+            output_frame.rfi_total += frame.rfi_total;
 
             // Move to next frame
             nframes += 1;
             mark_frame_empty(in_buf, unique_name.c_str(), frame_id);
             frame_id = (frame_id + 1) % in_buf->num_frames;
 
         } else {
+
+            timespec output_age = std::get<1>(frame.time) - std::get<1>(output_frame.time);
+            if (ts_to_double(output_age) > max_age) {
+                skipped_frame_counter.labels({std::to_string(output_frame.freq_id), "age"}).inc();
+                continue;
```

Comment:
I might be missing something but are you missing: 
```
mark_frame_empty(in_buf, unique_name.c_str(), frame_id);
frame_id = (frame_id + 1) % in_buf->num_frames;
```
?

Reply:
No. The logic is a little confusing, but I don't need that. I did need a `num_frames = 0` though. Or so I've figured out with Tristan.